tests/pass/tls/macos_tlv_atexit.rs

```diff
@@ -0,0 +1,34 @@
+//@only-target-darwin
+
+use std::ptr;
+use std::thread;
+
+extern "C" {
+    fn _tlv_atexit(dtor: unsafe extern "C" fn(*mut u8), arg: *mut u8);
+}
+
+fn register(dtor: extern "C" fn(*mut u8)) {
+    unsafe {
+        _tlv_atexit(dtor, ptr::null_mut());
+    }
+}
+
+extern "C" fn register_another(_: *mut u8) {
+    println!("registering another dtor");
+    register(run);
+}
+
+extern "C" fn run(_: *mut u8) {
+    println!("dtor was called");
+}
+
+fn main() {
+    thread::spawn(|| {
+        register(run);
+        register(run);
```

Comment:
This does not ensure that they are run in the right order, also wrt running newly registered dtors immediately.

Reply:
Sure!